functions/go/set-setter/main.go

```diff
@@ -0,0 +1,160 @@
+// This file will be processed and embedded to pluginator.
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/set-annotation/setters"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/api/wrappy"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+	"sigs.k8s.io/yaml"
+)
+
+type setSetterSpecs struct {
+	Setters []setSetterSpec `json:"setters,omitempty" yaml:"setters,omitempty"`
+}
+
+type setSetterSpec struct {
+	SetterName  string      `json:"setter_name,omitempty" yaml:"setter_name,omitempty"`
+	SetterValue interface{} `json:"setter_value,omitempty" yaml:"setter_value,omitempty"`
+}
+
+func setSetters(specs setSetterSpecs,
+	resMap resmap.ResMap,
+	pluginHelpers *resmap.PluginHelpers,
+	plugin *plugin) error {
+
+	err := plugin.Config(pluginHelpers, []byte{})
```

Comment:
You don't need to use a kustomize style plugin if you are not reusing an existing kustomize plugin. In this case I suppose you can directly use kyaml with your filters. So you can avoid importing kustomize/api deps.